src/number_theory/lucas_primality.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2020 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import data.zmod.basic
+import data.fintype.basic
+import field_theory.finite.basic
+import group_theory.order_of_element
+
+/-!
+# The Lucas test for primes.
+
+This file implements the Lucas test for primes (not to be confused with the Lucas-Lehmer test for
+Mersenne primes). A number `a` witnesses that `n` is prime if `a` has order `n-1` in the
+multiplicative group of integers mod `n`. This is checked by verifying that `a^(n-1) = 1 (mod n)`
+and `a^d ≠ 1 (mod n)` for any divisor `d | n - 1`. This test is the basis of the Pratt primality
+certificate.
+
+## TODO
+
+- Bonus: Show the reverse implication i.e. if a number is prime then it has a Lucas witness.
+  See [this stackexchange](https://math.stackexchange.com/a/59911/165144) answer for proof.
```

Comment:
I'm confused what you intend for this TODO? We have the results https://leanprover-community.github.io/mathlib_docs/ring_theory/integral_domain.html#units.is_cyclic and even the same proof via https://leanprover-community.github.io/mathlib_docs/group_theory/specific_groups/cyclic.html#is_cyclic_of_card_pow_eq_one_le already.

Reply:
Cool, I wasn't aware the proof was in mathlib already. All I intend for the TODO then is to use this fact that the multiplicative group mod p is cyclic for prime p to show that there exists an `a` (namely a generator of that cyclic group), such that the test passes for `a` (in other words, that `∀ q : ℕ, q.prime → q ∣ (p-1) → a^((p-1)/q) ≠ 1`)